src/vcpkg/vcpkgpaths.cpp

```diff
@@ -998,21 +998,94 @@ namespace vcpkg
                            Tools::GIT);
     }
 
-    ExpectedL<std::map<std::string, std::string, std::less<>>> VcpkgPaths::git_get_local_port_treeish_map() const
+    Optional<std::map<std::string, std::string, std::less<>>> VcpkgPaths::git_get_local_port_treeish_map(
+        DiagnosticContext& context) const
     {
-        const auto cmd = git_cmd_builder({}, {})
-                             .string_arg("-C")
-                             .string_arg(this->builtin_ports_directory())
-                             .string_arg("ls-tree")
-                             .string_arg("-d")
-                             .string_arg("HEAD")
-                             .string_arg("--");
+        auto git_exe = get_tool_exe(
+            Tools::GIT,
+            out_sink); // this should write to `context` but the tools cache isn't context aware at this time
+        Command prefix(git_exe);
+        prefix.string_arg("-c").string_arg("core.autocrlf=false");
+        RedirectedProcessLaunchSettings launch_settings;
+        launch_settings.working_directory.emplace(this->builtin_ports_directory());
+
+        std::string index_file;
+        {
+            Command cmd = prefix;
+            cmd.string_arg("rev-parse");
+            cmd.string_arg("--path-format=absolute");
+            cmd.string_arg("--git-path");
+            cmd.string_arg("index");
+
+            auto maybe_index_result = cmd_execute_and_capture_output(context, cmd, launch_settings);
+            if (auto index_output = check_zero_exit_code(context, maybe_index_result, git_exe))
+            {
+                auto trimmed = Strings::trim_end(*index_output);
+                auto last_slash = std::find(trimmed.rbegin(), trimmed.rend(), '/').base();
+                if (last_slash == trimmed.begin())
+                {
+                    context.report_error_with_log(
+                        *index_output, msgGitUnexpectedCommandOutputCmd, msg::command_line = cmd.command_line());
+                    context.report(
+                        DiagnosticLine{DiagKind::Note, msg::format(msgWhileGettingLocalTreeIshObjectsForPorts)});
+                    return nullopt;
+                }
+
+                index_output->resize(last_slash - trimmed.begin());
+                fmt::format_to(std::back_inserter(*index_output), "index.{}.tmp", get_process_id());
+                index_file = std::move(*index_output);
+            }
+            else
+            {
+                context.report(DiagnosticLine{DiagKind::Note, msg::format(msgWhileGettingLocalTreeIshObjectsForPorts)});
+                return nullopt;
+            }
+        }
+
+        auto& environment = launch_settings.environment.emplace();
+        environment.add_entry("GIT_INDEX_FILE", index_file);
```

Comment:
Does this file get cleaned up if `git write-tree` fails?

Reply:
Nope! Good catch on that bug.

Reply:
Fixed in https://github.com/microsoft/vcpkg-tool/pull/1616/commits/2cf695a0043dbb68053bb8eb8af466778342e766 ; note all paths in `write_git_tree` go through filesystem.remove